lib/index.js

```diff
@@ -1,21 +1,32 @@
 // Utility functions for ES6 Promises.
 
-module.exports = function(Promise) {
+module.exports = function(ParentPromise, smash) {
   "use strict";
 
-  if (!Promise) { Promise = global.Promise; }
+  if (!ParentPromise) { ParentPromise = global.Promise; }
   // Try to use Promise implementation from es6-shim if there isn't already one
   // installed.
-  if (!Promise) {
+  if (!ParentPromise) {
     try {
       require('es6-shim');
     } catch (e) {
-      throw new Error('No Promise implementation found. ' +
-        'Try installing the optional dependencies.');
+      throw new Error("No Promise implementation found. " +
+        "Install the optional dependencies to use es6-shim's Promises.");
     }
-    Promise = global.Promise;
+    ParentPromise = global.Promise;
   }
 
+  // create a new Promise subclass (this is less cumbersome in es6!)
+  var PrFunPromise = function PrFunPromise(exec) {
+    return ParentPromise.call(this, exec);
+  };
+  Object.setPrototypeOf(PrFunPromise, ParentPromise);
+  PrFunPromise.prototype = Object.create(ParentPromise.prototype);
+  PrFunPromise.prototype.constructor = PrFunPromise;
+
+  // sometimes we just need to smash things (sigh)
+  var Promise = smash ? ParentPromise : PrFunPromise;
```

Comment:
If possible, help the user out and test that subclassing will work. Or, maybe keep a list of known subclassable implementations in the README (in addition to es6-shim).